src/main/java/com/vadmack/mongodbtest/service/ProjectService.java

```diff
@@ -19,12 +23,27 @@
 
     private final ProjectRepository repository;
     private final SequenceGeneratorService sequenceGeneratorService;
+    private final PageableService pageableService;
 
     private final ModelMapper modelMapper = new ModelMapper();
 
-    public List<ProjectDto> findAll() {
-        return repository.findAll().stream().map(this::entityToDto)
-                .collect(Collectors.toList());
+    public List<ProjectDto> findList(
+            String name,
+            Optional<Integer> pageNumber,
```

Comment:
You can use `org.jetbrains.annotations.Nullable` annotation instead of `Optional`. IDEA can analyze such parameters and provide some warnings regarding the incorrect parameter usage

Additionally, the `name` can be `null` too.